teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +195,385 @@ In addition to grouping related resources, Tekton Results provides long term sto
+recommended by the [Kubernetes API conventions][subobjects]. 
+
+###### Alternatives
+* [Use Maps](#add-minimal-embedded-status---use-map)
+* [Collapse Fields](#add-minimal-embedded-status---collapse-fields)
+
+#### 2. Deprecate and Remove Full Embedded Status
+
+Deprecate and remove the old fields from `PipelineRunStatusFields` from the Beta API.
+
+```go
+type PipelineRunStatusFields struct {
+	...
+	TaskRunsStatus []PipelineRunTaskRunMinimalStatus `json:"taskRunsStatus,omitempty"`
+	RunsStatus []PipelineRunRunMinimalStatus `json:"runsStatus,omitempty"`
+	...
+}
+```
+
+This is a backwards incompatible change in Beta, therefore the fields will be deprecated
+and removed per out deprecation policy, as described in the [Beta API](#beta-api) section
+below.
+
+### Example
+
+This is an example `PipelineRun` status as provided in the [documentation][example]:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  ...
+spec:
+    ...
+status:
+    completionTime: "2020-05-04T02:19:14Z"
+    conditions:
+      - lastTransitionTime: "2020-05-04T02:19:14Z"
+        message: "Tasks Completed: 4, Skipped: 0"
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+    startTime: "2020-05-04T02:00:11Z"
+    taskRuns:
+      triggers-release-nightly-build:
+        pipelineTaskName: build
+        status:
+          completionTime: "2020-05-04T02:10:49Z"
+          conditions:
+            - lastTransitionTime: "2020-05-04T02:10:49Z"
+              message: All Steps have completed executing
+              reason: Succeeded
+              status: "True"
+              type: Succeeded
+          podName: triggers-release-nightly-build-pod
+          resourcesResult:
+            - key: commit
+              resourceRef:
+                name: git-source-triggers
+              value: 9ab5a1234166a89db352afa28f499d596ebb48db
+          startTime: "2020-05-04T02:05:07Z"
+          steps:
+            - container: step-build
+              imageID: docker-pullable://golang@sha256:a90f2671330831830e229c3554ce118009681ef88af659cd98bfafd13d5594f9
+              name: build
+              terminated:
+                containerID: docker://6b6471f501f59dbb7849f5cdde200f4eeb64302b862a27af68821a7fb2c25860
+                exitCode: 0
+                finishedAt: "2020-05-04T02:10:45Z"
+                reason: Completed
+                startedAt: "2020-05-04T02:06:24Z"
+```
+
+Taking the above example, this will be the new minimal `PipelineRun` status:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  ...
+spec:
+    ...
+status:
+    completionTime: "2020-05-04T02:19:14Z"
+    conditions:
+      - lastTransitionTime: "2020-05-04T02:19:14Z"
+        message: "Tasks Completed: 4, Skipped: 0"
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+    startTime: "2020-05-04T02:00:11Z"
+    taskRunsStatus:
```

Comment:
super nit: taskRunStatuses vs taskRunsStatus? (and same for runs?)

Reply:
fixed, thanks!